### DISCUSSION TITLE
Buggy(?) prefix handling in Espresso module

### DISCUSSION THREAD

**Discussion by hironori-kondo** [Quantum-Accelerators/quacc#2365] [General]
Hello @tomdemeyere, another series of questions for you on the Espresso module front:

1. In your `quacc.calculators.espresso.espresso.EspressoTemplate` implementation, the `_output_handler` overrides the user-specified `input_data` with the default `outkeys` generated by `quacc.calculators.espresso.utils.espresso_prepare_dir`. What was your reasoning for this decision? I ask because it also overrides any user-specified prefixes, which I see you've made room for elsewhere---this creates a discrepancy between the prefix in the Espresso-generated input/output files and the quacc results json file. 
2. In the various copying steps (both grid and standard), it seems that the implementation relies on the default prefix naming, so `copy_files`, etc. fails for user-specified prefixes. What was your rationale for this decision?

If the above seem like areas that should be repaired, I am happy to take a crack at them. It would be a good chance for me to work with more of the source code. I can get to the `grid_phonon_flow` patch after this, such that I can implement it with `copy_files` (should users want to disable the auto-relaxation).

**Comment by tomdemeyere** [comment]
@nerdwarrior No custom prefix or outdir with this interface, sorry! In fact anything related to file handling should not be touched, Quacc takes care!

**Reply by hironori-kondo** [reply]
To clarify, you mean the entire entire Espresso module, correct? I was confused by some non-default prefix handling, but it seems those are for specific patches? 

What of fildyn, etc. of ph.x? It seems `fildyn` is fixed to default but `fildvscf` is open for specification.

**Reply by tomdemeyere** [reply]
Yes the problem with this specific keyword (and some others?) is that if you specify it then you activate an additional functionality, therefore there is no default.

I am not sure this is ideal behaviour, there might be better way to deal with that

**Comment by tomdemeyere** [comment]
Btw is you wish to participate to the Espresso interface, (From your PR on ASE I assume you are already planning to add some EPW recipes?), but here some other things:

- There is the proposal about the xml reader, in the open issues

- Currently there is no way to run an Environ calculation using QuAcc (you need an additional "environ.in" file) it would be nice to allow that, in the static or relax jobs

**Reply by hironori-kondo** [reply]
Hi Tom---yes, I wanted to get some EPW working with quacc. Down the line, proper integration would be fantastic. Currently, I have plans for a specific flow, but EPW is particular with file i/o (hence my digging around). 

Noted re: XML reader, will take a look. I know less about Environ, but good to know! I'm not a programmer and not much of a computational chemist (four weeks ago, had never heard of ASE, VASP, or Espresso), but working on a related project and trying to get there. Grateful for your prompt & thorough replies!

**Reply by Andrew-S-Rosen** [reply]
Just wanted to chime in to say you're doing a great job for being so new! Thank you for your detailed reports and questions thus far, and thanks for trying out quacc!
